code/game/objects/items/weapons/storage/bluespace.dm

```diff
@@ -37,6 +37,12 @@
 		return // HOLY FUCKING SHIT WHY STORAGE CODE, WHY - pomf
 	if(istype(W, /obj/item/weapon/storage/backpack/holding/grinch))
 		return
+	if(istype(W, /obj/item/weapon/holder/diona)) // Dionas wearing bags of holding cause bagulooses too
```

Comment:
Instead of checking for a (specifically diona?) mob holder, you should use recursive_type_check() to recursively check the contents of the object being inserted for a BoH. Doing so should catch all edge cases instead of just this one.

Reply:
Thanks, applied the fix and it cut down code significantly while still working